### ISSUE TITLE
4.8.0 fails to compile

### ISSUE THREAD

**Issue by cole-wilson** [ORF-4450/RobotLib#13]
When I try to update to RobotLib 4.8.0 in the MaxSwerve test code, it errors with this message:
```
* Where:
Build file '/Users/cole/FRC/MaxSwerve/build.gradle' line: 93

* What went wrong:
Could not determine the dependencies of task ':jar'.
> Could not resolve all files for configuration ':runtimeClasspath'.
   > Could not find com.github.ORF-4450:RobotLib:4.8.0.
     Searched in the following locations:
       - file:/Users/cole/.m2/repository/com/github/ORF-4450/RobotLib/4.8.0/RobotLib-4.8.0.pom
       - https://repo.maven.apache.org/maven2/com/github/ORF-4450/RobotLib/4.8.0/RobotLib-4.8.0.pom
       - file:/Users/cole/wpilib/2024/maven/com/github/ORF-4450/RobotLib/4.8.0/RobotLib-4.8.0.pom
       - https://frcmaven.wpi.edu/artifactory/vendor-mvn-release/com/github/ORF-4450/RobotLib/4.8.0/RobotLib-4.8.0.pom
       - https://jitpack.io/com/github/ORF-4450/RobotLib/4.8.0/RobotLib-4.8.0.pom
     Required by:
         project :

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 1s
```
Also, when I went directly to JitPack to get the build, it also failed to build the jar with [this error log](https://jitpack.io/com/github/ORF-4450/RobotLib/v4.8.0/build.log).

**Comment by MoSadie** [comment]
Hi former team member here with unsolicited help.

Your error log seems to agree that jitpack failed to build RobotLib 4.8.0

It looks like with 2024 WPILib the library is targeting Java 17 features, right now jitpack is using Java 11. This can be changed in the jitpack.yml file. (Set it to 'openjdk17' iirc)

Once that change is committed you can try building that new commit on jitpack. If that works I'd recommend a new release or moving the tag and deleting the failed build on jitpack.

Hope any of that helps!

**Comment by Crossle86** [comment]
Sean, thanks for chiming in. Saved me from tracking you down :-) I will try your suggestion.

**Comment by Crossle86** [comment]
Well I updated to  openjdk17 in jitpack.yml and committed to the repo. Tried jitpack compile again, same error.

**Comment by cole-wilson** [comment]
Thank you for your help everyone!

**Comment by Crossle86** [comment]
Sean, I updated jitpack.yml again and it seems to be compiling now...BUT the version number on the resulting files is v1.8.0-gc8f9cdb-383 not 4.8.0 as it should be. Any help would be greatly appreciated as we are really pushing my understanding of how this all works.

**Comment by cole-wilson** [comment]
It looks like [this log](https://jitpack.io/com/github/cole-wilson/RobotLib/test/build.log) from a copy of the code I made has the correct version. Could you try incrementing the robotlib version to 4.8.1 and try again? I believe that might solve the problem.

If you go to the "commits" tab [here](https://jitpack.io/#ORF-4450/RobotLib) and build the latest one it works.

**Comment by MoSadie** [comment]
Looks like https://github.com/ORF-4450/RobotLib/commit/4c4a878e5bdf3bd634d128cd6e071bbe037c1434 compiles correctly without the v1.8 weirdness (not sure where that came from) if you make a release off that commit it should (hopefully) work.

I can look a little harder in a few hours when I get home from work, see if there's any ways to simplify this system.

**Comment by Crossle86** [comment]
I'm confused. Cole how did you do that test? Sean, that commit 4c4a878 appeard to fail when I tried it so I added the 
before_install:
  - sdk install java 17.0.8
  - sdk use java 17.0.8
jdk:
  - openjdk17

to the yml and then it compiled but had the -v1.8 stuff. Are you saying I don't need that sdk install step for 17? Or should I just do another release as cole suggested.

**Comment by cole-wilson** [comment]
So what I did was went to the jitpack link and went to the commits tab and chose the latest commit to build

**Comment by cole-wilson** [comment]
I think that the second update you made was unnecessary (although I don't think it hurt). I would try doing a new release and if it doesn't work you could roll back to the current step.

**Comment by Crossle86** [comment]
I just completed a new release to 4.8.1 and it compiled and the artifacts name is closed but not correct: v4.8.1 instead of 4.8.1

I agree the second update to jitpack.yml was not needed. I got confused by the jitpack builds and thought the first update to yml had failed to compile. In any case, we should be back to the same case as the first yml update. Should have worked.

So Cole you are suggesting I roll  back to 4c4a878 commit and try 4.8.0 again, after deleting the existing 4.8.0 ad 4.81 releases? That seems to make sense.

**Comment by Crossle86** [comment]
If I roll back,  what happens to the commits I made after that point? Will the be removed? Haven't rolled back before but have deleted releases...

Sean what do you think?

**Comment by cole-wilson** [comment]
You can't ever easily delete commits without using the command line git program, which can go wrong pretty easily. I would just attempt to delete the releases and try again. One thing to note is that the v4.8.1 release is still giving me the same error in VSCode.

**Comment by Crossle86** [comment]
Yes because the artifacts are named v4.8.1 and the robolib json has 4.8.1 (no v). You could test changing your robotlib.json.

I can delete the releases, but with no change in commits I would be at the same place I created 4.8.1 from.

**Comment by cole-wilson** [comment]
update: I just tried it again and v4.8.1 worked without changing anything. At this point I would just keep it as-is and skip the 4.8.0 release

**Comment by Crossle86** [comment]
Interesting. I just triggered a rebuild of 4.6.0 and it generated with a v, ie v4.6.0. If you look at the old build for 4.6.0 the artifact name is 4.6.0 as expected. Now 4.6.0 is set to jdk 11 and nothing changed in the source for the build from when it was originally compiled on jitpack. It should have worked the same as original on rebuild. But the v is there in todays rebuild  of 4.6.0. At this point I want to wait for Sean to weigh in on all this. I need to get ready to go to CHS.

**Comment by Crossle86** [comment]
As Cole said, 4.8.1 release is working, the version in RobotLib.json is 4.8.1 and it works though the artifact (file) names have v4.8.1 in them. I assumed the json version and the file name version would need to be the same but Jitpack is handling it apparently. Not sure at all what happened today but maybe I confused Jitpack  somehow.

Sean, pls take a look at the  generateJSON step of the build. It still fails but is easy to work around. If you can't fix it, I will just document the  manual update of robolib.json and leave it at that.

**Comment by cole-wilson** [comment]
closing this because it's technically fixed but still might need discussion

**Comment by MoSadie** [comment]
Sorry for the delay, work got busy.

On JitPack, all of the options to transition to other simpler services I found would be too complicated for a mid-season transition. (Two ones I researched were GitHub packages, which requires auth tokens that complicates setup, and Maven Central, which requires registering and proving ownership of domains before publishing, that's too much for a mid-build season change)

On the json not uploading I believe I have a fix for that, will test it later today.
